packaging/nix/flake.nix

```diff
@@ -2,55 +2,52 @@
   description = "umu universal game launcher";
 
   inputs = {
-    nixpkgs = {
-      type = "github";
-      owner = "NixOS";
-      repo = "nixpkgs";
-      ref = "nixpkgs-unstable";
-    };
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
   };
 
   outputs = {
     self,
     nixpkgs,
   }: let
-    umu-launcher-src = builtins.toPath "${self}/../../";
-
-    pkgs = import nixpkgs {
-      system = "x86_64-linux";
-      overlays = [self.overlays.default];
-    };
+    inherit (nixpkgs) lib;
 
-    version = "1.1.4";
-  in {
-    overlays.default = final: prev: let
-      py = prev.python3;
-    in {
-      umu-launcher = final.callPackage ./umu-launcher.nix {
-        inherit version;
-        umu-launcher = umu-launcher-src;
-        pyth1 = py;
-      };
+    # Utility function for producing consistent rename warning messages
+    rename = old: new: lib.warn "`${old}` has been renamed to `${new}`";
 
-      umu-run = final.callPackage ./umu-run.nix {
-        inherit version;
-        package = final.umu-launcher;
-      };
+    # Package sets supported by umu-launcher-unwrapped
+    # { system = pkgs; }
+    supportedPkgs =
+      lib.filterAttrs
+      (system: pkgs: builtins.elem system pkgs.umu-launcher-unwrapped.meta.platforms)
+      nixpkgs.legacyPackages;
 
-      umu = final.callPackage ./combine.nix {
+    # Use the current revision for the default version
+    version = self.dirtyShortRev or self.shortRev or self.lastModifiedDate;
```

Comment:
What's dirtyShortRev?

Also, if you want another fallback, you could add a static string like `"dirty"`.

Reply:
> What's dirtyShortRev?

`dirtyShortRev` is present when the flake has uncommitted changes. When present, it should be used instead of `shortRev` to indicate the that this version is different to `shortRev` in some unknown way.

> Also, if you want another fallback, you could add a static string like `"dirty"`.

There shouldn't be any need for another fallback, since `lastModifiedDate` is always present.

We could have a static string _instead of_ `lastModifiedDate` though. Maybe `"impure"`?

Reply:
I'd rather keep lastModifiedDate if it's always present.